### DISCUSSION TITLE
Using section-properties for the analysis of mass timber cross-sections

### DISCUSSION THREAD

**Discussion by jchkoch** [robbievanleeuwen/section-properties#437] [Ideas]
First of all fantastic package, I have found myself going back to it more and more when needing to analyze CompoundGeometry steel cross-sections. Of particular interest to me is how section-properties might be used to calculate the cross-sectional properties and a stress state of for example cross-laminated timber. Currently, section-properties seems able to compute the $(EI)_{eff}$ bending stiffness for both the major and minor axis if each layer of the CLT is modelled as a CompoundGeometry class, where two materials for alternating layers is used (i.e. $E_0$ and $E_90$ of a timber material).

The following gist shows a example 3-ply CLT section https://gist.github.com/jchkoch/ced0142406d635ba0bc76b88feff67e3.
Using a separate spreadsheet check the major axis $E_0$ bending stiffness (modulus-weighted as it were) seem to be correct. The minor axis $E_{90}$ bending stiffness seems to be too large; however, that could be due to the fact that most design methods of CLT tend to neglect the outer layers. I will be further checking into this and try to validate if this would be a approach to analysis CLT cross-sections.

My general question would first if section-properties is useful for this application (my initial test seems to indicate yes) and whether adding support for anisotropic materials would be interesting for this package. My current thought would be to develop a specific geometry class for CLT (e.g. for the definition of layers) and a material class which could be applied to the geometry object.

**Comment by robbievanleeuwen** [comment]
Hi @jchkoch thanks for your kind words, I'm glad you are getting good mileage out of `sectionproperties`!

I am open to having a new timber/CLT [section library](https://sectionproperties.readthedocs.io/en/stable/user_guide/geometry.html#section-library) for easily creating these cross-sections. I assume this would be similar to the current libraries, which are just methods that return a `Geometry` or `CompoundGeometry` object with assigned material properties. If you're interested in contributing, I'd be happy for you to create a pull-request. You could use the existing code as a template, e.g. the [`concrete_circular_section`](https://sectionproperties.readthedocs.io/en/stable/_modules/sectionproperties/pre/library/concrete_sections.html#concrete_circular_section) method, to create new CLT/timber methods in a new file in the `src/sectionproperties/pre/library` directory. I'd be happy to help if you ran into any barriers :)

**Comment by Grefron** [comment]
Hi, I am an engineering working with fibre-reinforced polymers in bridge construction and I have just being playing about with this great package and I would like to express my interest in getting anisotropic materials working properly. 
In stead of modelling all the layers (plies) in the structure, which would result in a very complex geometry, I take a slightly different approach where I first calculate the laminate properties from ply properties using the classical lamination theory and use those properties in the analysis. I have overridden the Material class however, to take the shear modulus directly, otherwise  this would be incorrectly calculated. Although my initial results show the output is quite OK, I still need to do more tests and validation.

I do not have the skills honestly to contribute much, apart from code that implements the lamination theory is there is interest.

One thing I

**Comment by TLCFEM** [comment]
> First of all fantastic package, I have found myself going back to it more and more when needing to analyze CompoundGeometry steel cross-sections. Of particular interest to me is how section-properties might be used to calculate the cross-sectional properties and a stress state of for example cross-laminated timber. Currently, section-properties seems able to compute the (EI)eff bending stiffness for both the major and minor axis if each layer of the CLT is modelled as a CompoundGeometry class, where two materials for alternating layers is used (i.e. E0 and E90 of a timber material).
> 
> The following gist shows a example 3-ply CLT section https://gist.github.com/jchkoch/ced0142406d635ba0bc76b88feff67e3. Using a separate spreadsheet check the major axis E0 bending stiffness (modulus-weighted as it were) seem to be correct. The minor axis E90 bending stiffness seems to be too large; however, that could be due to the fact that most design methods of CLT tend to neglect the outer layers. I will be further checking into this and try to validate if this would be a approach to analysis CLT cross-sections.
> 
> My general question would first if section-properties is useful for this application (my initial test seems to indicate yes) and whether adding support for anisotropic materials would be interesting for this package. My current thought would be to develop a specific geometry class for CLT (e.g. for the definition of layers) and a material class which could be applied to the geometry object.

Laminated material in this case CLT and anisotropic material are two different concepts in general.
Assuming the out-of-paper axis is one of the principal axes, and it is fully decoupled from two in-plane axes (not true in most cases), then you have a plane material with two principal axes which may not be orthogonal to each other, it is still necessary to account for material orientation, which means there is a need to incorporate a rotation matrix.

For CLT, often it is modeled via orthogonal material models. When the material stiffness is a matrix, the underlying math needs to be reformulated.

With the current codebase, I am afraid that one can at most extend it to very simple orthogonal case, and the material has to be aligned with major and minor axes. I don't think it is universally useful.

It needs quite some work, if this is the direction to go, maybe start with simple orthogonal case.
